src/components/Pagination.tsx

```diff
@@ -0,0 +1,132 @@
+import {useState, useEffect, useRef} from 'react';
+import '../styles/Pagination.scss';
+
+type PaginationProps = {
+  totalRecords?: number,
+  recordsLimit?: number,
+  totalPages?: number,
+  onPageChange?: () => {}
+}
+
+const Pagination = ({ totalRecords, recordsLimit, onPageChange, totalPages = 0 }:PaginationProps) => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [hasLeftSpill, setHasLeftSpill] = useState(false);
+  const [hasRightSpill, setHasRightSpill] = useState(false);
+  const paginationWrapper:any = useRef();
+  const pageNeighbours = 1;
+
+  const range = (start:number, end:number) => {
+    const range = [];
+
+    for(let i = start; i <= end; i++) {
+      range.push(<a className="pagination-nav-item" href={`/${i}`} data-target={i} onClick={(e) => {onHandleClick(e)}} key={i}>{i}</a>);
+    }
+
+    return range;
```

Comment:
Maybe it can be written one line: return new Array(end-start).fill(null).map((_,i) => (<a className="pagination-nav-item" href={`/${i}`} data-target={i} onClick={(e) => {onHandleClick(e)}} key={i}>{i}</a>))